### DISCUSSION TITLE
Debug logging

### DISCUSSION THREAD

**Discussion by AlUlkesh** [vladmandic/sdnext#372] [Q&A]
Hi @vladmandic ,

The image browser extension has a much needed debugging option using logger. But it seems something in your fork turns it or maybe the handlers off. Do you perhaps have a suggestion on how to integrate this better?

The relevant code is this, executed at the beginning of the extension:
```
logger = logging.getLogger(__name__)
...
    if level_value >= debug_level_value:
        logger_mode = logging.DEBUG
...
logger.setLevel(logger_mode)
if (logger.hasHandlers()):
    logger.handlers.clear()
console_handler = logging.StreamHandler()
console_handler.setLevel(logger_mode)
formatter = logging.Formatter(f'%(asctime)s image_browser.py: %(message)s', datefmt='%Y-%m-%d-%H:%M:%S')
console_handler.setFormatter(formatter)
logger.addHandler(console_handler)
...
    file_handler = logging.FileHandler(log_file)
    file_handler.setLevel(logger_mode)
    file_handler.setFormatter(formatter)
    logger.addHandler(file_handler)
```

**Comment by vladmandic** [comment]
first thing that launch sequence does is to disable logging below INFO level as many packages are extremely verbose.
E.g., just Pillow was logging hundreds of lines on first import.

The issue is not visible in A1111 since there is no log handler there, so all those messages were going nowhere and by the time extension (such as image browser) loads, all that storm has passed.

I can try disable debug logging on startup, but re-enable it later so extensions can use it.

btw, good to see you here :)

**Comment by AlUlkesh** [comment]
Thanks. I'm glad you're actively working on this fork. Must be daunting, now that it gains popularity. Good luck.
